File: stage_5/SparkX/backend/app/models/entities/Employee.go
package entities

import "time"

// Employee структура сущности сотрудника
type Employee struct {
	ID         int64      `json:"ID"`         // идентификатор
	Lastname   string     `json:"lastname"`   // фамилия
	Firstname  string     `json:"firstname"`  // имя
	Middlename string     `json:"middlename"` // отчество
	Email      string     `json:"email"`      // почтовый адрес
	Login      string     `json:"login"`      // логин
	Password   string     `json:"password"`   // пароль
	Avatar     string     `json:"avatar"`     // аватар
	Online     bool       `json:"online"`     // онлайн
	LastIn     *time.Time `json:"lastIn"`     // последнее время входа
}

File: stage_5/SparkX/frontend/src/js/src/models/entities/employee.ts
import EmployeeKanban from "./employeeKanban";

export default class Employee {
    public ID: number;
    public lastname: string;
    public firstname: string;
    public middlename: string;
    public email: string;
    public login: string;
    public avatar: string;
    public lastIn: Date;
    public online: boolean;
    public password: string;

    constructor(
        id: number,
        lastname: string,
        firstname: string,
        middlename: string,
        email: string,
        login: string,
        avatar: string,
        lastIn: string,
        online: boolean
    ) {
        this.ID = id;
        this.lastname = lastname;
        this.firstname = firstname;
        this.middlename = middlename;
        this.email = email;
        this.login = login;
        this.avatar = avatar;
        this.lastIn = new Date(lastIn);
        this.online = online;
    }

    public static fullName(employee: Employee | EmployeeKanban): string {
        return `${employee.lastname} ${employee.firstname} ${employee.middlename}`;
    }
}